packages/frontend/src/features/semanticViewer/runners/SemanticViewerResultsRunner.ts

```diff
@@ -112,119 +156,98 @@ export class SemanticViewerResultsRunner implements IResultsRunner {
         );
     }
 
-    getDimensions(): VizIndexLayoutOptions[] {
-        return this.dimensions.map((dimension) => ({
-            reference: dimension.name,
-            axisType: getVizIndexTypeFromDimensionType(dimension.type),
-            dimensionType: getDimensionTypeFromSemanticLayerFieldType(
-                dimension.type,
-            ),
-        }));
-    }
-
-    getMetrics(): VizValuesLayoutOptions[] {
-        return this.metrics.map((metric) => ({
-            reference: metric.name,
-            aggregation: metric.aggType || VIZ_DEFAULT_AGGREGATION,
-        }));
+    async getPivotedVisualizationData(
+        query: SemanticLayerQuery,
+    ): Promise<PivotChartData> {
+        if (!!query.pivot?.index.length || !!query.pivot?.values.length) {
+            return {
+                results: [],
+                indexColumn: undefined,
+                valuesColumns: [],
+                columns: [],
+            };
+        }
+        return this.runPivotQuery(query);
     }
 
     getPivotQueryDimensions(): VizIndexLayoutOptions[] {
+        // the same as pivotChartIndexLayoutOptions
         return this.dimensions.map((dimension) => ({
             reference: dimension.name,
-            axisType: getVizIndexTypeFromDimensionType(dimension.type),
+            axisType: getVizIndexTypeFromSemanticLayerFieldType(dimension.type),
             dimensionType: getDimensionTypeFromSemanticLayerFieldType(
                 dimension.type,
             ),
         }));
     }
 
     getPivotQueryMetrics(): VizValuesLayoutOptions[] {
+        // returns empty for sql runner because there's no metrics
         return this.metrics.map((metric) => ({
             reference: metric.name,
             aggregation: metric.aggType || VIZ_DEFAULT_AGGREGATION,
-            aggregationOptions: [],
         }));
     }
 
     getPivotQueryCustomMetrics(): VizCustomMetricLayoutOptions[] {
```

Comment:
mostly logic pulled from sql-runner. it works for semantic layer too, we just don't use it yet

Reply:
Cool

Reply:
But do you anticipate this always working the same way on both pages? If they always share this, any time we change one, we'll be changing the other.